### ISSUE TITLE
Property in subclass goes missing due to wrong OpenAPI specification?

### ISSUE THREAD

**Issue by MarcMil** [swagger-api/swagger-core#3063]
I am using Swagger Core 2.0.6
I do have the problem that a property, ```additionalPropertyWhichShouldBeThere``` does not seem to get properly placed into the OpenAPI specification.
```
package testPackage;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "SubClass")
public class SubClass extends BaseClass {
	@Schema(required = false, description = "Test")
	public int additionalPropertyWhichShouldBeThere = -1;

}
```

```
package testPackage;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "test", discriminatorProperty="type")
@JsonTypeInfo(
		  use = JsonTypeInfo.Id.NAME, 
		  include = JsonTypeInfo.As.PROPERTY, 
		  property = "type")
		@JsonSubTypes({ 
		  @Type(value = SubClass.class, name = "SubClass")
		})
public class BaseClass {

	@Schema(required = true, description = "Type", example="AndroidDeviceRequirements")
	public String type;
	
	public BaseClass(String type) {
		this.type = type;
	}
	
	public BaseClass() {
	}

}
```

I used swagger core to obtain the OpenAPI specification:
```
		OpenAPI openAPI = reader.read(classes);
		Yaml.mapper().setSerializationInclusion(JsonInclude.Include.NON_NULL);
		String s = Json.pretty().writeValueAsString(openAPI);
```
Part of the specification:
```
  "components" : {
    "schemas" : {
      "BaseClass" : {
        "required" : [ "type" ],
        "type" : "object",
        "properties" : {
          "type" : {
            "type" : "string",
            "description" : "Type",
            "example" : "AndroidDeviceRequirements"
          }
        },
        "description" : "test",
        "discriminator" : {
          "propertyName" : "type"
        }
      },
      "SubClass" : {
        "required" : [ "type" ],
        "type" : "object",
        "properties" : {
          "additionalPropertyWhichShouldBeThere" : { / <--- this property should be in the allOf below
            "type" : "integer",
            "description" : "Test",
            "format" : "int32"
          }
        },
        "description" : "SubClass",
        "allOf" : [ {
          "$ref" : "#/components/schemas/BaseClass"
           <--- in addition to the $ref, there should be additional properties
        } ]
      }
    }
  }
```
The Swagger code generator generates code of the SubClass without ```additionalPropertyWhichShouldBeThere```:
```
@Schema(description = "SubClass")
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2018-12-21T20:47:57.468+01:00[Europe/Berlin]")public class SubClass extends BaseClass {
  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    return super.equals(o);
  }

  @Override
  public int hashCode() {
    return java.util.Objects.hash(super.hashCode());
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class SubClass {\n");
    sb.append("    ").append(toIndentedString(super.toString())).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
```

According to https://stackoverflow.com/a/35916573/1120897 it seems that that property should be listed in the ```allOf``` part. When I fix that manually, the code generator for Java generates the correct code. 

Am I missing something? Maybe a wrong annotation?

Thank you very much in advance for any hint!

**Comment by MarcMil** [comment]
Probably a duplicate of https://github.com/swagger-api/swagger-core/issues/3030?

**Comment by frantuma** [comment]
thanks for reporting this, also duplicates #3197 and #3030; fixed in #3262
